CIP-Registration/CIP-Registration.md

```diff
@@ -0,0 +1,101 @@
+---
+CIP: TODO
+Title: Catalyst Registration transaction metadata format
+Authors: Sebastien Guillemot <sebastien@emurgo.io>, Rinor Hoxha <rinor.hoxha@iohk.io>, Mikhail Zabaluev <mikhail.zabaluev@iohk.io>
+Comments-URI: TODO
+Status: Draft
+Type: Standards
+Created: 2020-01-05
+License: CC-BY-4.0
+---
+
+## Abstract
+
+Cardano uses a sidechain for its treasury system. One needs to "register" to participate on this sidechain by submitting a registration transaction on the mainnet chain. This CIP details the registration transaction format.
+
+## Motivation
+
+Cardano uses a sidechain for its treasury system ("Catalyst"). One of the desirable properties of this sidechain is that even if its safety is compromised, it doesn't cause loss of funds on the main Cardano chain. To achieve this, instead of using your wallet's recovery phrase on the sidechain, we need to use a brand new "voting key".
+
+However, since 1 ADA = 1 vote, a user needs to associate their mainnet ADA to their new voting key. This can be achieved through a registration transaction.
+
+We therefore need a registration transaction that serves two purposes:
+
+1. Registers a "voting key" to be included in the sidechain
+2. Associates mainnet ADA to this voting key
+
+## Specification
+
+### Voting key format
+
+A voting key is simply an ED25519 key. How this key is created is up to the wallet.
+
+### Associating stake with a voting key
+
+Recall: Cardano uses the UTXO model so to completely associate a wallet's balance with a voting key (i.e. including enterprise addresses), we would need to associate every payment key to a voting key individually. Although there are attempts at this (see [CIP8](../CIP-0008/CIP-0008.md)), the resulting data structure is a little excessive for on-chain metadata (which we want to keep small)
+
+Given the above, we choose to only associate staking keys with voting keys. Since most Cardano wallets only use base addresses for Shelley wallet types, in most cases this should perfectly match the user's wallet.
+
+### Registration metadata format
+
+A Catalyst registration transaction are a regular Cardano transaction with a specific transaction metadata associated with it
+
+Notably, there should be two entries inside the metadata map:
+
+Voting key registration
```

Comment:
I'm not sure if the prose here helps. I was thinking of just inlining these comments into the CDDL definition and then getting rid of this example, but I had two concerns:

1) People may have a hard time visualizing the CDDL format if they aren't familiar with CDDL
2) cardano-wallet and similar tools use this format with the 0x prefix to render metadata so it's not bad to show it here